Mathlib/Probability/Distributions/Exponential.lean

```diff
@@ -0,0 +1,251 @@
+
+lemma exponentialPdf_eval_pos {r : ℝ} : ∀ᵐ x : ℝ ∂ volume , x < 0 →
+    exponentialPdf r x = 0 := by
+  simp only [exponentialPdf_eq]
+  exact ae_of_all _ (fun x hx ↦ by simp [not_le.mpr hx])
+
+lemma exponentialPdf_eval_neg {r : ℝ} : ∀ᵐ x : ℝ ∂ volume, (x ∈ {x | 0 ≤ x} →
+    exponentialPdf r x =
+    ENNReal.ofReal (r * rexp (-(r * x)))) := by
+  simp only [exponentialPdf_eq]
+  exact ae_of_all _ (fun x (hx : 0 ≤ x) ↦ by rw [if_pos hx])
+
+lemma antiDeriv_tendsto_zero {r : ℝ} (hr : 0 < r) :
+    Tendsto (fun x ↦ -1/r * exp (-(r * x))) atTop (𝓝 0) := by
+  rw [← mul_zero (-1/r)]
+  apply Tendsto.mul
+  · exact tendsto_const_nhds
+  · apply tendsto_exp_neg_atTop_nhds_0.comp
+    exact (tendsto_const_mul_atTop_of_pos hr).2 tendsto_id
+
+open Measure
+
+lemma lintegral_exponentialPdfReal_eq_one (r : ℝ) (hr : 0 < r) :
+    ∫⁻ x, exponentialPdf r x = 1 := by
+  rw [lintegral_split (exponentialPdf r) 0, ← ENNReal.toReal_eq_one_iff]
+  have leftSide : ∫⁻ x in {x | x < 0}, exponentialPdf r x = 0 := by
+    rw [set_lintegral_congr_fun (isOpen_gt' 0).measurableSet
+      exponentialPdf_eval_pos, lintegral_zero]
+  have rightSide : ∫⁻ x in {x | x ≥ 0}, exponentialPdf r x
+      = ∫⁻ x in {x | x ≥ 0}, ENNReal.ofReal (r * rexp (-(r * x))) := by
+    exact set_lintegral_congr_fun isClosed_Ici.measurableSet exponentialPdf_eval_neg
+  simp only [leftSide, add_zero]
+  rw [rightSide, ENNReal.toReal_eq_one_iff, ←ENNReal.toReal_eq_one_iff]
+  rw [← integral_eq_lintegral_of_nonneg_ae (ae_of_all _ (fun _ ↦ by positivity))]
+  · simp only [ge_iff_le, restrictₗ_apply]
+    have IntegrOn : IntegrableOn (fun x ↦ rexp (-(r * x))) (Ioi 0) := by
+      simp only [← neg_mul, exp_neg_integrableOn_Ioi 0 hr]
+    rw [integral_mul_left, Ici_def, integral_Ici_eq_integral_Ioi,
+        integral_Ioi_of_hasDerivAt_of_tendsto' (fun _ _ ↦ antiDeriv_expDeriv_pos' hr) IntegrOn
+        (antiDeriv_tendsto_zero hr)]
+    field_simp
+  · exact ((measurable_id'.const_mul r).neg.exp.const_mul r).stronglyMeasurable.aestronglyMeasurable
+
+/-- The Pdf of the exponential Distribution integrates to 1-/
+@[simp]
+lemma lintegral_exponentialPdf_eq_one (r : ℝ) (hr : 0 < r) :
+    ∫⁻ x, exponentialPdf r x = 1 :=
+  lintegral_exponentialPdfReal_eq_one r hr
+
+end ExponentialPdf
+
+open MeasureTheory
+
+/-- Measure defined by the exponential Distribution -/
+noncomputable
+def expMeasure (r : ℝ) : Measure ℝ :=
+  volume.withDensity (exponentialPdf r)
+
+instance instIsProbabilityMeasureExponential (r : ℝ) [Fact (0 < r)] :
+    IsProbabilityMeasure (expMeasure r) where
+  measure_univ := by simp [expMeasure, lintegral_exponentialPdf_eq_one r Fact.out]
+
+section ExponentialCdf
+
+/-- CDF of the exponential Distribution -/
+noncomputable
+def exponentialCdfReal (r : ℝ) : StieltjesFunction :=
+    cdf (expMeasure r)
+
+lemma expCdf_eq_integral (r : ℝ) [Fact (0 < r)] : exponentialCdfReal r
+    = fun x ↦ ∫ x in Iic x, exponentialPdfReal r x := by
```

Comment:
The name of the object is exponentialCdfReal, not expCdf. Also it's better to add an argument that to write an equality of functions:
```suggestion
lemma exponentialCdfReal_eq_integral (r : ℝ) [Fact (0 < r)] (x : ℝ) :
    exponentialCdfReal r x = ∫ x in Iic x, exponentialPdfReal r x := by
```